java/code/src/com/suse/manager/webui/services/pillar/MinionCustomInfoPillarGenerator.java

```diff
@@ -37,25 +38,30 @@
 
     public static final MinionCustomInfoPillarGenerator INSTANCE = new MinionCustomInfoPillarGenerator();
 
+    public static final String CATEGORY = "custom_info";
+
     /**
      * Generates pillar data containing CustomInfo information of the passed minion
      * @param minion the minion server
      * @return the SaltPillar containing the pillar data
      */
     @Override
-    public Optional<SaltPillar> generatePillarData(MinionServer minion) {
+    public Optional<Pillar> generatePillarData(MinionServer minion) {
         if (minion.getCustomDataValues().isEmpty()) {
             return Optional.empty();
         }
```

Comment:
Deleting all Custom Info values should delete the pillar.
I don't see it handled neither here nor in MinionPillarFileManager.
(before it was handled by deleting the pillar file in MinionPillarFileManager)

Reply:
ACK